src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -561,13 +562,60 @@ checkout_base_tree (RpmOstreeSysrootUpgrader *self,
                        &self->tmprootfs_dfd, error))
     return FALSE;
 
-  /* build a centralized rsack for it, since we need it in a few places */
-  self->rsack = rpmostree_get_refsack_for_root (self->tmprootfs_dfd, ".", error);
+  rpmostree_output_task_end ("done");
+  return TRUE;
+}
+
+static gboolean
+load_base_rsack (RpmOstreeSysrootUpgrader *self,
+                 GCancellable             *cancellable,
+                 GError                  **error)
+{
+  /* go through all the deployments and check if one of them has the base rev */
```

Comment:
First, this is another thing that is going to break with livefs, right?

But more importantly, we [added the base rpmdb](https://github.com/projectatomic/rpm-ostree/pull/1142) before for this reason, we should use that, no?

Reply:
Ahh good point re. livefs, I should add a check for that.